lib/manager/npm/post-update/lerna.ts

```diff
@@ -84,12 +86,12 @@ export async function generateLockFiles(
     cmd.push(`${lernaClient} install ${cmdOptions}`);
     let lernaVersion: string;
     try {
-      const pJson = JSON.parse(await platform.getFile('package.json'));
+      const pJson = JSON.parse(await platform.getFile(packageJson));
```

Comment:
I wasn't actually sure if there was a reason for this to delegate to a git show command, rather than read the file from the filesystem?

Reply:
We're in the process of removing as many of these git.* commands to replace with fs.*

Reply:
That said -- now that a `PackageFile` is passed to this method, rather than just `lernaClient: string`, does it even really need to do this at all? `PackageFile` has a `deps` property (no idea if it's set at this stage)

Reply:
Here's an alternative approach that doesn't require re-loading this file:
```
diff --git a/lib/manager/npm/post-update/lerna.ts b/lib/manager/npm/post-update/lerna.ts
index 5a628aa34..7fa155ea7 100644
--- a/lib/manager/npm/post-update/lerna.ts
+++ b/lib/manager/npm/post-update/lerna.ts
@@ -3,7 +3,6 @@ import { quote } from 'shlex';
 import { join } from 'upath';
 import { logger } from '../../../logger';
 import { ExecOptions, exec } from '../../../util/exec';
-import { getFile } from '../../../util/git';
 import { PackageFile, PostUpdateConfig } from '../../common';
 import { getNodeConstraint } from './node-version';
 import { optimizeCommand } from './yarn';
@@ -86,12 +85,10 @@ export async function generateLockFiles(
     cmd.push(`${lernaClient} install ${cmdOptions}`);
     let lernaVersion: string;
     try {
-      const pJson = JSON.parse(await getFile(packageJson));
-      lernaVersion =
-        (pJson.dependencies && pJson.dependencies.lerna) ||
-        (pJson.devDependencies && pJson.devDependencies.lerna);
+      lernaVersion = lernaPackageFile.deps.find((d) => d.depName === 'lerna')
+        .currentValue;
     } catch (err) {
-      logger.warn(`Could not detect lerna version in ${packageJson}`);
+      logger.warn(`Could not detect lerna version in ${packageJson}: ${err}`);
     }
     if (!lernaVersion || !semver.validRange(lernaVersion)) {
       lernaVersion = 'latest';
```
Would that work for all scenarios?

Reply:
Yes, if the packageFile has `deps` then that could be a cleaner solution

Reply:
The try catch can be removed then too

Reply:
@rarkins @viceice 👍 I pushed a commit that does that. Since docker preCommands don't seem to be captured by snapshots, I'm unsure how to test this though.
Happy to hop on to Slack to discuss this if I get an invite :)

Reply:
This lack of test is also probably why coverage isn't reached anymore.